pkg/host/internal/network/network.go

```diff
@@ -196,3 +200,93 @@ func (n *network) GetNetDevLinkSpeed(ifaceName string) string {
 
 	return fmt.Sprintf("%s Mb/s", strings.TrimSpace(string(data)))
 }
+
+// GetDeviceParam returns devlink parameter for the device as a string, if the parameter has multiple values
+// then the function will return only first one from the list.
+func (n *network) GetDevlinkDeviceParam(pciAddr, paramName string) (string, error) {
+	funcLog := log.Log.WithValues("device", pciAddr, "param", paramName)
+	funcLog.V(2).Info("GetDevlinkDeviceParam(): get device parameter")
+	param, err := n.netlinkLib.DevlinkGetDeviceParamByName(consts.BusPci, pciAddr, paramName)
+	if err != nil {
+		funcLog.Error(err, "GetDevlinkDeviceParam(): fail to get devlink device param")
+		return "", err
+	}
+	if len(param.Values) == 0 {
+		funcLog.Error(fmt.Errorf("param %s has no value", paramName),
+			"GetDevlinkDeviceParam(): error")
+	}
+	var value string
+	switch param.Type {
+	case nl.DEVLINK_PARAM_TYPE_U8, nl.DEVLINK_PARAM_TYPE_U16, nl.DEVLINK_PARAM_TYPE_U32:
+		var valData uint64
+		switch v := param.Values[0].Data.(type) {
+		case uint8:
+			valData = uint64(v)
```

Comment:
just as an FYI: you can use `fallthrough` keyword here and in the second case and have assignment only on the last case.

Reply:
you can't use `fallthrough` in type switches. Also, option with single `case uint8, uint16, uin32:`  will not work here because `v` will have `interface{}` type inside the case branch, and we want `v` to have a specific uint type.

Reply:
ack ! makes sense.